test/e2e/router_compression_test.go

```diff
@@ -198,38 +198,40 @@ func TestRouterCompressionOperation(t *testing.T) {
 		t.Fatalf("failed to get route: %v", err)
 	}
 
+	routeHost := getRouteHost(t, r, ic.Name)
+
 	// curl to canary, without the Accept-Encoding header set to gzip
-	if err := testContentEncoding(t, client, r, false, ""); err != nil {
+	if err := testContentEncoding(t, client, routeHost, false, ""); err != nil {
 		t.Error(err)
 	}
 
 	// curl to canary, WITH the Accept-Encoding header set to gzip
-	if err := testContentEncoding(t, client, r, true, "gzip"); err != nil {
+	if err := testContentEncoding(t, client, routeHost, true, "gzip"); err != nil {
 		t.Error(err)
 	}
 }
 
 // testContentEncoding makes a call to the provided route, adds a gzip content header if addHeader is true, and
 // compares the returned Content-Encoding header to the given expectedContentEncoding.  If expectedContentEncoding
 // is the same as the returned Content-Encoding header, then the test succeeds.  Otherwise it fails.
-func testContentEncoding(t *testing.T, client *http.Client, route *routev1.Route, addHeader bool, expectedContentEncoding string) error {
+func testContentEncoding(t *testing.T, client *http.Client, routeHost string, addHeader bool, expectedContentEncoding string) error {
 	t.Helper()
 
 	if err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
-		header, code, err := getHttpHeaders(client, route, addHeader)
+		header, code, err := getHttpHeaders(client, routeHost, addHeader)
 
 		if err != nil {
-			t.Logf("GET %s failed: %v, retrying...", route.Spec.Host, err)
+			t.Logf("GET %s failed: %v, retrying...", routeHost, err)
 			return false, nil
 		}
 		if code != http.StatusOK {
-			t.Logf("GET %s failed: status %v, expected %v, retrying...", route.Spec.Host, code, http.StatusOK)
+			t.Logf("GET %s failed: status %v, expected %v, retrying...", routeHost, code, http.StatusOK)
 			return false, nil // retry on 503 as pod/service may not be ready
 		}
 
 		contentEncoding := header.Get("Content-Encoding")
 		if contentEncoding != expectedContentEncoding {
-			return false, fmt.Errorf("compression error: expected %q, got %q for %s route", expectedContentEncoding, contentEncoding, route.Name)
+			return false, fmt.Errorf("compression error: expected %q, got %q for %s route", expectedContentEncoding, contentEncoding, routeHost)
```

Comment:
I think this is fine, but curious if you meant to change `route.Name` to the hostname of the route?

Reply:
It was an intentional change; I needed `routeHost` elsewhere in the function, and it seemed simpler to use it in this log message rather than pass in both `routeHost` and `route`.

Reply:
Ack, sounds good to me.